src/main/webapp/script.js

```diff
@@ -117,102 +115,171 @@ function initMap() {
-      imgElement.onerror = function() {
-        this.onerror = null;
-        this.src = '';
-      }
-      resElement.appendChild(imgElement);
+    const imgElement = document.getElementById("restaurant-image");
+    imgElement.src = url;
+    imgElement.onerror = function() {
+      this.onerror = null;
+      this.src = '';
     }
+  }
 
-    resListElement.appendChild(resElement);
-
-    resElement.addEventListener('click', function() {
-      map.panTo(location);
-      map.setZoom(18);
+  map.panTo(location);
+  map.setZoom(18);
 
-      updateRestaurantInfo(restaurants[index], markers[index]);
-      selectedElement.classList.remove("selected");
-      this.classList.add("selected");
-      selectedElement = this;
+  let params = {
+    api: 1,
+    destination: name,
+    destination_place_id: placeId
+  }
+  const dirUrl = new URL('https://www.google.com/maps/dir/');
+  Object.keys(params).forEach(key => dirUrl.searchParams.append(key, params[key]));
 
-      markers[selectedIndex].setIcon(blueIconUrl);
-      markers[index].setIcon(orangeIconUrl);
-      selectedIndex = index;
-    });
+  const directionElement = document.getElementById("restaurant-dir");
+  const whatsappElement = document.getElementById("restaurant-whatsapp-share");
+  const telegramElement = document.getElementById("restaurant-telegram-share");
 
-    let restaurantMarker = new google.maps.Marker({
-      map: map,
-      anchorPoint: new google.maps.Point(0, -29),
-      position: location,
-      title: name,
-      icon: {
-        url: blueIconUrl
-      }
-    });
+  directionElement.href = dirUrl;
+  whatsappElement.href = `https://wa.me/?text=${encodeURIComponent(dirUrl)}%0ALet's%20eat%20at%20${encodeURI(name)}!`;
+  telegramElement.href = `https://t.me/share/url?url=${encodeURIComponent(dirUrl)}&text=Let's%20eat%20at%20${encodeURI(name)}!`;
 
-    markers.push(restaurantMarker);
 
-    if(index == 0) {
-      selectedIndex = 0;
-      selectedElement = resElement;
-      resElement.classList.add("selected");
-      restaurantMarker.setIcon(orangeIconUrl);
-      map.panTo(location);
-      map.setZoom(18);
-      updateRestaurantInfo(restaurants[index], markers[index]);
+  restaurantMarker && restaurantMarker.setMap(null);
+  restaurantMarker = new google.maps.Marker({
+    map: map,
+    anchorPoint: new google.maps.Point(0, -29),
+    position: location,
+    title: name,
+    icon: {
+      url: blueIconUrl
     }
+  });
+  updateRestaurantInfo(restaurants[index], restaurantMarker);
+}
+
+function nextRestaurant() {
+  restaurantIndex = (restaurantIndex + 1) % restaurants.length;
+  updateRestaurantDetails(restaurantIndex);
+}
+
+function prevRestaurant() {
+  restaurantIndex = (restaurantIndex === 0) ? restaurants.length - 1 : restaurantIndex - 1;
+  updateRestaurantDetails(restaurantIndex);
+}
+
+function setRestaurantTab(hidden) {
+  const resultContainerElement = document.getElementById("result-container");
+  if(hidden) {
```

Comment:
Missing space before the bracket (

Reply:
Actually ignore this, half the code has space before bracket and half don't. Let's fix the linting in a separate PR